### ISSUE TITLE
Notifications: Cha-Ching push notification sound isn't used

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-android#1076]
**Describe the bug**

The "Cha-Ching!" sound for new order notifications isn't working. I get the push notification for a new order, but no sound is made.

The only way to get the notification to make a sound is to switch from the app-provided sound to a different system notification sound, or to reinstall the WooCommerce app.

Although the issue is resolved by reinstalling the app from the Play Store, I'm reporting it because it has come up in user reviews (including from external beta testers) and not all users experiencing the issue may try reinstalling.

**To Reproduce**
1. Go to the app Settings > Manage notifications.
2. Confirm "New order alerts" is enabled.
3. Tap on the "New order alerts" setting and confirm that under "Sound" it's set to "App provided sound."
4. Close the app. On the web, complete a new order on your store.
5. Wait for the new order push notification. Result: The notification arrives but no sound is made.

Note: This issue went away when I reinstalled the app, so this may be related to updating from an older version of the app or another action I have taken previously.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: moto e5 play
- Android version: 8.1.0
- WooCommerce Android version: 1.8-rc-1

**Comment by nbradbury** [comment]
@rachelmcr I've tried without success to reproduce this but I'm unable to. I've seen users report this, though, so I know there's a problem somewhere.

Before I dive in deeper, I want to make sure the lack of sound isn't because there's already a Woo push notification in the system bar, because we only play a sound when there's no notification already showing.

**Comment by rachelmcr** [comment]
> I want to make sure the lack of sound isn't because there's already a Woo push notification in the system bar, because we only play a sound when there's no notification already showing.

Nope, there weren't any push notifications in the system bar. I tested different notifications (confirming that I get a sound with other push notifications from the app) but I always cleared the notification before trying again.

I'll keep an eye on my notifications to see if I can catch this happening again and trace it back to more reliable repro steps.

**Comment by rachelmcr** [comment]
A few testing notes for this issue:

* Clearing app data or reinstalling the app makes the notification sound work again. (Clearing just the app cache does not resolve it.)
* Once the notification sound is working again, updating the app version does not cause the issue to reappear.
* Switching stores in the app does not cause the issue to reappear.
* Receiving a review notification and then a purchase notification does not cause the issue to reappear.

So far I haven't hit on the right step(s) to make the notification sound stop working, but over time as I use the app eventually the sound stops working and I have to clear data/reinstall to get it working again.

I have seen several reports of this in the Play Store from beta and non-beta app users:

> Somehow no ca ching sound for Android

> I just removed a star as the cash register notification disappeared.

> Notification of a new request is not working after the last update.

**Comment by zwiesman31** [comment]
User here seemingly has same issue: 2209162-zen

**Comment by rachelmcr** [comment]
I noticed this came up again in a recent app review:

> Nice app, but since updates the checkout katsjinggg notice sound is gone, that is a pity, i liked that...

Galaxy S8, Android 9, app version 3.6

**Comment by tucson-az** [comment]
I was having the same issue, which led me here.  However, I just figured out what the issue was (for *me*, anyway...)

I had my store set up in demo mode and kept testing different aspects of it. Sometimes the "cha-ching" sound happened, other times it didn't.  I found that if I kept placing 'test orders' from the same 'user', it only notified me the first time.  If I opened a new incognito session and input a new 'buyer', I would get the "cha-ching" notification again.  

Maybe you all already knew this... just thought I'd share in case someone didn't.

**Comment by AmandaRiu** [comment]
@tucson-az 

> Maybe you all already knew this... just thought I'd share in case someone didn't.

That's very helpful, thank you.

**Comment by anitaa1990** [comment]
Could be related to #1926

**Comment by rossanafmenezes** [comment]
Another user having the same issue here 3147921-Zen

**Comment by sarahfg** [comment]
From the above issue: 
Moto g7 play, XT1952-1
Android 10
Android security Patch level: 1st June 2020

**Comment by erricgunawan** [comment]
Still from 3147921-zen (which is followed up on 3202663-zen):

>  I've reinstalled the app. It made a very slight vibration when I received an order but still no cha-ching sound.

> No special battery saver options or other things relating to the set up and It's one order at a time that comes in.

> The problem seemed to occur after I upgraded to Android 10, which was around the end of June, possibly before, although I am unsure if this is the reason or whether it could have been an earlier update of woo commerce.

**Comment by eduardozulian** [comment]
From app reviews on Jan 19, 2021:

> The sound of the notifications does not work, I installed it in 3 phones and it continues with the same problem, they should fix that error, because if the application does not fulfill its function

Reference: p1611108784003700-slack-C013AAPA4G0

**Comment by rachelmcr** [comment]
From app reviews on May 8, 2021:

> [...] notification sound stopped working for me. When it did work you can't change the sound to something else either.

Device: Samsung Galaxy S21+ 5G
Android version: Android 11 (SDK 30)

Reference: p1620495855048500-slack-C013AAPA4G0

**Comment by jostnes** [comment]
Seems like this is happening again: 4429635-zd-woothemes

> "Hi, just installed your app on Galaxy s52, and i cant hear the "cha ching " alert anymore"

Have asked user for more info and will follow up in ticket.

**Comment by reginabally** [comment]
Adding the app log section for the report mentioned in the above comment. It shows that the app did receive the push notification for new order but it didn't make a "cha-ching" sound.

```
[Nov-04 16:00 WP d] API Dispatching action: NotificationAction-REGISTER_DEVICE
[Nov-04 16:00 WP d] API Dispatching action: NotificationAction-REGISTERED_DEVICE
[Nov-04 16:00 NOTIFS d] Sending FCM token to our remote services: ce9CA3FbWLM:APA91bHmbq3D2J7U9Lv7RImdmPxiJz-DJ6SopyNJ-IyEaGyCrl26wb3EM0ig-YSId12vpA6cd6mS_aPxez_sKcsRWP3jrE2g1RCmNMWcx0BhX6b9wjKr21MxkjgGZZVA3aHiTq-s2Gaw
[Nov-04 16:00 UTILS i] 🔵 Tracked: push_notification_received, Properties: {"notification_note_id":4907143871,"notification_type":"NEW_ORDER","push_notification_token":"ce9CA3FbWLM:APA91bHmbq3D2J7U9Lv7RImdmPxiJz-DJ6SopyNJ-IyEaGyCrl26wb3EM0ig-YSId12vpA6cd6mS_aPxez_sKcsRWP3jrE2g1RCmNMWcx0BhX6b9wjKr21MxkjgGZZVA3aHiTq-s2Gaw","blog_id":183043675,"is_wpcom_store":false,"is_debug":false}
```

**Comment by hichamboushaba** [comment]
@reginabally I checked the logs in the ticket, and they don't have the part that was added and that we need: `Notification channel already created`, the cause is that this part is logged only on app launch, and the log was overwritten (maybe we need to move it to the moment of showing the notification 🤔, but that's another story), for now, can you please ask the user to force close the app, then re-open it, and go directly to the support screen to share the logs?

**Comment by reginabally** [comment]
Thanks, @hichamboushaba! The user followed your instructions and shared with us the app log. I can see the `Notification channel already created` log entry:

```
[Nov-06 23:46 NOTIFS i] Notification channel already created with the following attributes: NotificationChannel{mId='wooandroid_notification_channel_order_id', mName=New order alerts, mDescription=, mImportance=3, mBypassDnd=false, mLockscreenVisibility=-1000, mSound=content://settings/system/notification_sound, mLights=false, mLightColor=0, mVibration=null, mUserLockedFields=24, mFgServiceShown=false, mVibrationEnabled=false, mShowBadge=true, mDeleted=false, mGroup='null', mAudioAttributes=AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_UNKNOWN flags=0x800 tags= bundle=null, mBlockableSystem=false, mAllowBubbles=-1, mImportanceLockedByOEM=false, mImportanceLockedDefaultApp=false, mOriginalImp=-1000, mParent=null, mConversationId=null, mDemoted=false, mImportantConvo=false}
[Nov-06 23:46 NOTIFS i] Notification channel already created with the following attributes: NotificationChannel{mId='wooandroid_notification_channel_review_id', mName=Product review alerts, mDescription=, mImportance=3, mBypassDnd=false, mLockscreenVisibility=-1000, mSound=content://settings/system/notification_sound, mLights=false, mLightColor=0, mVibration=null, mUserLockedFields=0, mFgServiceShown=false, mVibrationEnabled=false, mShowBadge=true, mDeleted=false, mGroup='null', mAudioAttributes=AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_SONIFICATION flags=0x800 tags= bundle=null, mBlockableSystem=false, mAllowBubbles=-1, mImportanceLockedByOEM=false, mImportanceLockedDefaultApp=false, mOriginalImp=-1000, mParent=null, mConversationId=null, mDemoted=false, mImportantConvo=false}
[Nov-06 23:46 NOTIFS i] Notification channel already created with the following attributes: NotificationChannel{mId='wooandroid_notification_channel_general_id', mName=General, mDescription=, mImportance=3, mBypassDnd=false, mLockscreenVisibility=-1000, mSound=content://settings/system/notification_sound, mLights=false, mLightColor=0, mVibration=null, mUserLockedFields=0, mFgServiceShown=false, mVibrationEnabled=false, mShowBadge=true, mDeleted=false, mGroup='null', mAudioAttributes=AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_SONIFICATION flags=0x800 tags= bundle=null, mBlockableSystem=false, mAllowBubbles=-1, mImportanceLockedByOEM=false, mImportanceLockedDefaultApp=false, mOriginalImp=-1000, mParent=null, mConversationId=null, mDemoted=false, mImportantConvo=false}
```

The `mSound` parameter is having a different value than what I'm having on my Android device. It looks like the app was using the default system notification. Let me ask the user to uninstall the app and reinstall it again.

**Comment by hichamboushaba** [comment]
Thanks @reginabally, this is the solution we support right now.
Sadly the Android SDK doesn't allow to reset the channel from the API, which was a choice that the Android team took, to give the user full control over it, but it doesn't work well for all cases.
If we receive a lot of issues regarding this, there is another solution, it's not pretty straightforward, but doable, but if the number of tickets is limited, I personally would prefer keeping this as they are right now, and we can ask users to uninstall/re-install the app if they messed with the settings.

**Comment by erricgunawan** [comment]
From 4570149-zen:

> I have an issue with new order tune ( alarm ) , before i had a unique and strong sound when i get new order but unfortunately when i update my woocomerce this tune disapper and i didnt find it at all , even i looked for it in sound setting but its not available .

> Notification setting checked but unfortunately i didnt find that unique tune for woocomerce ( there are many tunes but the original one for new order is not there ) , so i think with your last update something went wrong for this tune , you can check on your update 

~Asked device info and will update once it's provided.~

WooCommerce version: 8.0 
Android Device: Samsung Galaxy Note 9 ( SM N96F/DS )
Android Version:  10

**Comment by erricgunawan** [comment]
[4-star Android app review](https://play.google.com/console/u/0/developers/7957760354032996428/app/4973018994742253387/user-feedback/review-details?reviewId=gp%3AAOqpTOHTyEQGHrgBZk_xOP_JaVXvdb9jlJFrM6ucenIZCQnWTHYaxwVsrHN4zkgk-s9ubNvktRBvOvEcDN1aqvM&corpus=PUBLIC_REVIEWS):

> Very excellent, the latest update, but the only problem is that the distinctive default tone has been deleted when a new order arrives

Device: Samsung Galaxy Note9
Device language: Arabic
App version code: 262
App version name: 8.0
Android version: Android 10 (SDK 29)

**Comment by hichamboushaba** [comment]
@erricgunawan since the logs on the ticket don't have the part about notifications details, we need to either ask the user to re-launch the app then post the logs directly, to make sure they are included.
But IMO, we should go forward and ask them to uninstall then reinstall the app, I'm pretty sure they just removed the sound accidentally, and sadly it's not a thing that we can easily fix from our side.

**Comment by erricgunawan** [comment]
@hichamboushaba Sorry I missed your reply! 😞

I'll keep that in mind for future tickets... which I happen to have another one here 😅

4641935-zen 

> OS: 11
> DeviceName: samsung SM-A505G

They shared a log but I don't see the `Notification channel already created` there.

I followed your advice asking them to re-launch the app and share the logs. I'll leave a note for the next steps.

Thank you!

**Comment by rossanafmenezes** [comment]
User came back with the new report:

```
[Dec-30 01:44 NOTIFS i] Notification channel already created with the following attributes: NotificationChannel{mId='wooandroid_notification_channel_order_id', mName=Alertas de nuevo pedido, mDescription=, mImportance=4, mBypassDnd=false, mLockscreenVisibility=-1000, mSound=content://media/external/audio/media/71183?title=Shrink%20Ray&canonical=1, mLights=false, mLightColor=0, mVibration=null, mUserLockedFields=34, mFgServiceShown=false, mVibrationEnabled=true, mShowBadge=true, mDeleted=false, mGroup='null', mAudioAttributes=AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_UNKNOWN flags=0x800 tags= bundle=null, mBlockableSystem=false, mAllowBubbles=-1, mImportanceLockedByOEM=false, mImportanceLockedDefaultApp=false, mOriginalImp=-1000, mParent=null, mConversationId=null, mDemoted=false, mImportantConvo=false}
[Dec-30 01:44 NOTIFS i] Notification channel already created with the following attributes: NotificationChannel{mId='wooandroid_notification_channel_review_id', mName=Alertas de reseña de producto, mDescription=, mImportance=3, mBypassDnd=false, mLockscreenVisibility=-1000, mSound=content://settings/system/notification_sound, mLights=false, mLightColor=0, mVibration=null, mUserLockedFields=0, mFgServiceShown=false, mVibrationEnabled=false, mShowBadge=true, mDeleted=false, mGroup='null', mAudioAttributes=AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_SONIFICATION flags=0x800 tags= bundle=null, mBlockableSystem=false, mAllowBubbles=-1, mImportanceLockedByOEM=false, mImportanceLockedDefaultApp=false, mOriginalImp=-1000, mParent=null, mConversationId=null, mDemoted=false, mImportantConvo=false}
[Dec-30 01:44 NOTIFS i] Notification channel already created with the following attributes: NotificationChannel{mId='wooandroid_notification_channel_general_id', mName=General, mDescription=, mImportance=3, mBypassDnd=false, mLockscreenVisibility=-1000, mSound=content://settings/system/notification_sound, mLights=false, mLightColor=0, mVibration=null, mUserLockedFields=0, mFgServiceShown=false, mVibrationEnabled=false, mShowBadge=true, mDeleted=false, mGroup='null', mAudioAttributes=AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_SONIFICATION flags=0x800 tags= bundle=null, mBlockableSystem=false, mAllowBubbles=-1, mImportanceLockedByOEM=false, mImportanceLockedDefaultApp=false, mOriginalImp=-1000, mParent=null, mConversationId=null, mDemoted=false, mImportantConvo=false}
```

**Comment by hichamboushaba** [comment]
@rossanafmenezes so he is using a custom sound, the default sound that the app configures has this format: `mSound=android.resource://com.woocommerce.android/....`.

Let's ask them to uninstall then reinstall the app, then maybe we can explain to him that he should avoid changing the sound from the system settings, since we can't restore it for them.
I think it's safe to do so for all cases, without having to go through the request for new logs 🤔.

**Comment by rossanafmenezes** [comment]
Hi @hichamboushaba

I asked them to reinstall the app 😁 but forgot to mention it here. Sorry about that

**Comment by mdrockwell** [comment]
> Despite updating to the latest version of JetPack, which addressed the app notification issue I reported, I'm still experiencing the problem. I asked my host to white-list all JetPack's IPs, but this did not fix the issue. I am receiving notifications from the app when new orders come in, but there is no sound. I have enabled and allowed all notifications for my device, as per your advice in our last email communication.

6575865-zd-woothemes

I suggested they reinstall the app, but that didn't do the trick for them.

**Comment by hichamboushaba** [comment]
> 6575865-zd-woothemes
>
> I suggested they reinstall the app, but that didn't do the trick for them.

Thanks @mdrockwell for the report, and sorry for the late reply, I was AFK, and I think the rest of the team missed the ping.
Unfortunately, I don't think there is much we can do here to investigate, we are still positive this is a device behavior (and probably caused by the user's actions), so if the user (or another user) comes again, let's try these checks:
1. Ask them to reinstall the app.
2. Ask for the app's logs collected after the reinstallation.
3. Confirm the device is not silent.

**Comment by gs1623** [comment]
@hichamboushaba -- Hi there. I'm the user who reported this issue, and I am still experiencing it. I have already completed step one, and I can confirm the device is not on silent. So the only thing that is left to do is delete, reinstall, and provide you with the app's logs if that helps.

**Comment by hichamboushaba** [comment]
Thanks @gs1623 for letting me know.

> So the only thing that is left to do is delete, reinstall, and provide you with the app's logs if that helps.

Yes, this would be helpful, but please don't update the logs here, instead create another ticket, and provide the logs in the ticket, the support team can help you in finding the logs if needed, then please ask them to share the logs with me.

Thanks again.

**Comment by gs1623** [comment]
> Thanks @gs1623 for letting me know.
> 
> > So the only thing that is left to do is delete, reinstall, and provide you with the app's logs if that helps.
> 
> Yes, this would be helpful, but please don't update the logs here, instead create another ticket, and provide the logs in the ticket, the support team can help you in finding the logs if needed, then please ask them to share the logs with me.
> 
> Thanks again.

-----

Thanks for your response.

Don't worry, I'm aware this is a public post, so I will reopen the ticket I raised about the issue and send the logs privately there. I'll ask the support team to forward it to you.

I appreciate your response. I've been monitoring this post for a solution, so once I send the logs, hopefully you'll get to the bottom of it.

**Comment by hichamboushaba** [comment]
Let's close this now that #10327 has been implemented, if other users complain, let's show them how they can [restore](https://github.com/woocommerce/woocommerce-android/pull/10341) the sound from the app settings.

**Comment by hichamboushaba** [comment]
We were able to identify the root cause of the issues, and the above PR while helping, it doesn't solve it, so re-opening this.

**Comment by hichamboushaba** [comment]
This was fixed by #11095
